packages/anthropic/src/anthropic-messages-language-model.ts

```diff
@@ -210,6 +214,42 @@ export class AnthropicMessagesLanguageModel implements LanguageModelV2 {
       baseArgs.max_tokens = maxOutputTokens + thinkingBudget;
     }
 
+    let modifiedTools = tools;
+    let modifiedToolChoice = toolChoice;
+
+    if (anthropicOptions?.webSearch) {
+      const webSearchTool: any = {
+        type: 'web_search_20250305',
+        name: 'web_search',
+        ...(anthropicOptions.webSearch.maxUses && {
+          max_uses: anthropicOptions.webSearch.maxUses,
+        }),
+        ...(anthropicOptions.webSearch.allowedDomains && {
+          allowed_domains: anthropicOptions.webSearch.allowedDomains,
+        }),
+        ...(anthropicOptions.webSearch.blockedDomains && {
+          blocked_domains: anthropicOptions.webSearch.blockedDomains,
+        }),
+        ...(anthropicOptions.webSearch.userLocation && {
+          user_location: {
+            type: anthropicOptions.webSearch.userLocation.type,
+            country: anthropicOptions.webSearch.userLocation.country,
+            ...(anthropicOptions.webSearch.userLocation.city && {
+              city: anthropicOptions.webSearch.userLocation.city,
+            }),
+            ...(anthropicOptions.webSearch.userLocation.region && {
+              region: anthropicOptions.webSearch.userLocation.region,
+            }),
+            ...(anthropicOptions.webSearch.userLocation.timezone && {
+              timezone: anthropicOptions.webSearch.userLocation.timezone,
+            }),
+          },
+        }),
+      };
```

Comment:
do we need this more convoluted pattern? undefined values in the leaves might be fine and will get pruned when json stringify is called